docs/rules/generator-star-spacing.md

```diff
@@ -75,6 +75,25 @@ An example of shorthand configuration:
 "generator-star-spacing": ["error", "after"]
 ```
 
+Additionally, this rule allows further configurability via overrides per function type.
+
+* `named` provides overrides for named functions
+* `anonymous` provides overrides for anonymous functions
+* `method` provides overrides for class methods or property function shorthand
+* `static` provides overrides for static class methods
```

Comment:
Given that the `"before"` option is currently ignored for methods anyway, could we just not ignore it for static methods? That way people can use the `method` option for both.